configure.py

```diff
@@ -2402,5 +2403,56 @@ def create_building_system(args):
     generate_compdb('compile_commands.json', ninja, args.buildfile, selected_modes)
 
 
+def configure_for_mode(mode, args):
+    # TODO: use Ninja multi-config
+    source_dir = os.path.realpath(os.path.dirname(__file__))
+    build_dir = os.path.join(source_dir, 'build', mode)
+
+    settings = {
+        'CMAKE_BUILD_TYPE': mode,
+        'CMAKE_CXX_COMPILER': args.cxx,
+        'CMAKE_C_COMPILER': args.cc,
+        'CMAKE_C_FLAGS': args.user_cflags,
+        'CMAKE_EXE_LINKER_FLAGS': semicolon_separated(args.user_ldflags),
+        'CMAKE_EXPORT_COMPILE_COMMANDS': 'ON',
+        'CMAKE_EXPORT_NO_PACKAGE_REGISTRY': 'ON',
+    }
+    if args.date_stamp:
+        settings['Scylla_DATE_STAMP'] = args.date_stamp
+
+    cmake_command = ['cmake']
+    cmake_command += [f'-D{var}={value}' for var, value in settings.items()]
+    cmake_command += ['-G', 'Ninja',
+                      '-B', build_dir,
+                      '-S', source_dir]
+    subprocess.check_call(cmake_command, shell=False, cwd=source_dir)
+    return build_dir, os.path.join(build_dir, 'build.ninja')
+
+
+def configure_with_cmake(args):
+    ninja = find_ninja()
+    with open('build.ninja', 'w', encoding='utf-8') as main_build_file:
+        ninja_opts = ''
+        if args.verbose:
+            ninja_opts += ' --verbose'
+        main_build_file.write(textwrap.dedent(f'''
+          rule ninja
+            command = {ninja} {ninja_opts} -C $dir $target
+            restat = 1
+        '''))
+
+        all_modes = ['debug', 'release', 'dev', 'sanitize', 'coverage']
+        for mode in args.selected_modes or all_modes:
+            build_dir, build_file = configure_for_mode(mode.capitalize(), args)
+            main_build_file.write(textwrap.dedent(f'''
+              build {mode}-build: ninja {build_file}
+                dir = {build_dir}
+                target = scylla
+              '''))
+
+
 if __name__ == '__main__':
-    create_building_system(args)
+    if args.with_cmake:
+        configure_with_cmake(args)
+    else:
+        create_building_system(args)
```

Comment:
s/building_system/build_system

Ah I see, this is already the name of this method. A bit strange, but fine.

Reply:
sure, will change. build system is more correct.

Reply:
i created #15932. as this change might take longer to add the compatibility targets.